tutorials/nservicebus-101/lesson-2/tutorial.md

```diff
@@ -0,0 +1,182 @@
+    
+`SendLocal(object message)` is a method that is available on the `IEndpointInstance` interface, as we are using here, and also on the `IMessageHandlerContext` interface, which we saw when we were defining our message handler. The *Local* part means that we are not sending to an external endpoint (in a different process) so we intend to handle the message in the same endpoint that sent it. Using `SendLocal()`, we don't have to do anything special to tell the message where to go.
+
+NOTE: In the [next lesson](../lesson-3/) we'll learn how to send messages between separate endpoints, and how to control where the messages go.
+
+Because `SendLocal()` returns a `Task`, we need to be sure to `await` it properly.
+ 
+In order to use our `RunLoop` method, we need to replace the previous Console operations with a call to the new method in the `AsyncMain` method:
+
+snippet:AddRunLoopToAsyncMain
+
+
+### Running the solution
+
+Now we can run the solution. Whenever we type `placeorder` on the console, a command message is sent and then processed by a handler class in the same project.
+
+```no-highlight
+INFO  ClientUI.Program Enter 'placeorder' to place an order, or 'quit' to quit.
+placeorder
+INFO  ClientUI.Program Sending PlaceOrder command, OrderId = 1ec12713-f86d-43e1-a41f-16aba19dd6f7
+INFO  ClientUI.Program Enter 'placeorder' to place an order, or 'quit' to quit.
+INFO  ClientUI.PlaceOrderHandler Received PlaceOrder, OrderId = 1ec12713-f86d-43e1-a41f-16aba19dd6f7
+placeorder
+INFO  ClientUI.Program Sending PlaceOrder command, OrderId = 9260d6ce-6f3e-4b00-9edb-0dfb905afee8
+INFO  ClientUI.Program Enter 'placeorder' to place an order, or 'quit' to quit.
+INFO  ClientUI.PlaceOrderHandler Received PlaceOrder, OrderId = 9260d6ce-6f3e-4b00-9edb-0dfb905afee8
+```
+
+Note how after sending a message, the prompt from `ClientUI.Program` is displayed _before_ the `ClientUI.PlaceOrderHandler` acknowledges receipt of the message. This is because, rather than calling the `Handle` method as a direct method call, the message is sent asynchronously, and then control immediately returns to the `RunLoop`, which repeats the prompt. It isn't until a bit later, when the message is received and processed, that we see the `Received PlaceOrder` notification.
+
+
+## Summary
+
+In this lesson we learned about messages, message assemblies, and message handlers. We created a message and a handler and we used `SendLocal()` to essentially send the message to ourselves.
+
+In the next lesson, we'll create a second messaging endpoint, move our message handler over to it, and then configure the ClientUI to send the message to the new endpoint. We'll also be able to observe what happens when we send messages while the receiver endpoint is offline.
```

Comment:
General question - since locally sending messages is kind of a special case, do we want to encourage it? It is very prominent in this tutorial and only in 3rd lesson we explain that this is not how one should do it...